SolidLabResearch/loama
# Refactor: make controller modular

- Rewritten the controller to a class based package
- Moved code depending on third-party libraries to their separate class, so these are swappable with custom one's

## How do I test this PR?
- Select a file (I've mostly used the README file)
- In the right column, click the "Edit" button. This should open a drawer. 
  This drawer should contain a table with all the subjects who have access to the file
- Click the "New Subject" to give a new subject access. This will again open a new smaller drawer where you can select the subject type
- Select "WebId" and give a webId which already has been added to the ACL file/table (e.g. your own webId)
- This should give an error in the console + a toast notification that the webId has been added previously
- Now use a webId which isn't present in the ACL file/table
- Click on the "Create" button, the Drawer should close & the new webId should be visible in the table
- Now choose one of the entries in the table & hit the "Edit" button at the end of the row
- Grant a new permission by checking the checkbox in the new drawer that should've opened.
- When you close this drawer. You'll see that your changes are reflected in the table (give it a second to reload the data)
- You can do multiple grants/revokes at once in the "edit subject" drawer
  - If you do them too fast after each other the inrupt SDK will freak out and skip some changes to the ACL file which will result in a desync between the ACL & the index.json


depends on #16 
closes #8 
Does a bit of work for #19, the resource access is removed when all the grants are revoked
